ext/semian/simple_integer.c

```diff
@@ -5,13 +5,11 @@
 #include <sys/shm.h>
 #include "types.h"
 #include "util.h"
+#include "sysv_semaphores.h"
 
 static const rb_data_type_t semian_simple_integer_type;
 
-static semian_simple_integer_shared_t* get_value(VALUE self) {
-  semian_simple_integer_t *res;
-  TypedData_Get_Struct(self, semian_simple_integer_t, &semian_simple_integer_type, res);
-
+static semian_simple_integer_shared_t* get_value(semian_simple_integer_t* res) {
```

Comment:
This is a great example of where C++ function overloading makes a lot of sense... :(